[FIX] CI 중 마이 페이지 lint 오류 빌드 중단

## Summary

GitHub Actions CI에서 `pnpm lint` 단계에서 오류가 발생하여 빌드가 중단되었습니다.

<img width="540" alt="Image" src="https://github.com/user-attachments/assets/26d66fa4-ebdd-4bdd-b209-8a2ac1054add" />

## Description

### 상세 로그

```
- src/pages/my/_components/myReportModal.tsx(10,40): error TS6133: 'open' is declared but its value is never read.
- src/pages/my/_components/myShortsCard.tsx(4,31): error TS2307: Cannot find module './myreportModal' or its corresponding type declarations.
- src/pages/my/_components/myVideoCard.tsx(5,31): error TS2307: Cannot find module './myreportModal' or its corresponding type declarations.
- ELIFECYCLE  Command failed with exit code 2.
- Error: Command "pnpm run build" exited with 2
```

- [ ] `myReportModal`에서 사용되지 않는 open 변수를 제거해야 합니다.
- [ ] `myShortsCard`에서 임포트하고 있는 './myreportModal' 경로의 대소문자 오류를 수정해야 합니다.
- [ ] `myVideoCard`에서 임포트하고 있는 './myreportModal' 경로의 대소문자 오류를 수정해야 합니다.

```
// guide
// import { MyReportModal } from './myreportModal' 을 다음으로 수정해야 합니다.
import { MyReportModal } from './myReportModal'
```

<img width="600" alt="Image" src="https://github.com/user-attachments/assets/e31899b1-ec6f-4f0b-941a-52d8d50df126" />

--- a/frontend/src/components/Pagination.tsx
+++ b/frontend/src/components/Pagination.tsx
@@ -18,9 +18,9 @@ const Pagination = ({ totalItems, itemCountPerPage, currentPage, onChangePage }:
     const noPrev = startPage == 1
 
     useEffect(() => {
-        if (currentPage >= startPage + 5 && !noNext) setStartPage(currentPage)
-        else if (currentPage <= startPage - 1 && !noPrev) setStartPage(currentPage - 4)
-    }, [currentPage])
+        if (currentPage >= startPage + 5 && !noNext) setStartPage(currentPage) // 다음 페이지 리스트
+        else if (currentPage <= startPage - 1 && !noPrev) setStartPage(currentPage - 4) // 이전 페이지 리스트
+    }, [noPrev, noNext, startPage, currentPage])
 
     return (
         <div className="flex items-center gap-[16px]">

--- a/frontend/src/pages/my/_components/myReportModal.tsx
+++ b/frontend/src/pages/my/_components/myReportModal.tsx
@@ -3,11 +3,10 @@ import Modal from '../../../components/Modal'
 
 interface MyReportModalProps {
     title: string
-    open: boolean
     setOpen: (open: boolean) => void
 }
 
-export const MyReportModal = ({ title, open, setOpen }: MyReportModalProps) => {
+export const MyReportModal = ({ title, setOpen }: MyReportModalProps) => {
     const navigate = useNavigate()
     const getReport = () => {
         setOpen(false)

--- a/frontend/src/pages/my/_components/myShortsCard.tsx
+++ b/frontend/src/pages/my/_components/myShortsCard.tsx
@@ -1,7 +1,7 @@
 import { useState } from 'react'
 import type { Shorts } from '../../../types/profile'
 import { formatKoreanNumber, formatRelativeTime } from '../../../utils/format'
-import { MyReportModal } from './myreportModal'
+import { MyReportModal } from './myReportModal'
 
 interface MyShortsCardProps {
     shorts: Shorts
@@ -28,7 +28,7 @@ export default function MyShortsCard({ shorts }: MyShortsCardProps) {
                     </div>
                 </div>
             </div>
-            {open && <MyReportModal title={shorts.title} open={open} setOpen={setOpen} />}
+            {open && <MyReportModal title={shorts.title} setOpen={setOpen} />}
         </>
     )
 }

--- a/frontend/src/pages/my/_components/myVideoCard.tsx
+++ b/frontend/src/pages/my/_components/myVideoCard.tsx
@@ -2,7 +2,7 @@ import type { Video } from '../../../types/profile'
 import { useState } from 'react'
 
 import { formatKoreanNumber, formatRelativeTime } from '../../../utils/format'
-import { MyReportModal } from './myreportModal'
+import { MyReportModal } from './myReportModal'
 
 interface MyVideoCardProps {
     video: Video
@@ -26,7 +26,7 @@ export default function MyVideoCard({ video }: MyVideoCardProps) {
                     </div>
                 </div>
             </div>
-            {open && <MyReportModal title={video.title} open={open} setOpen={setOpen} />}
+            {open && <MyReportModal title={video.title} setOpen={setOpen} />}
         </>
     )
 }

